pkg/cri/server/container_status.go

```diff
@@ -59,7 +59,11 @@ func (c *criService) ContainerStatus(ctx context.Context, r *runtime.ContainerSt
 			imageRef = repoDigests[0]
 		}
 	}
-	status := toCRIContainerStatus(container, spec, imageRef)
+	status, err := toCRIContainerStatus(ctx, container, spec, imageRef)
+	if err != nil {
```

Comment:
see comment below this should always be successful

Reply:
is there any reason why we don't want to return error here?

Reply:
More of a vague memory + gut feeling than anything else on the not verbose get containerstatus() call.. We have had a lot of history with timing problems here. 

If an rm of the container was running in async with the status request... the start of the status request may succeed in retrieving status from the store but the by the time you go back to get the spec that call may fail.. Something that would have previously resulted in returning stopped state will now return error.  There were also some issues surrounding unknown state, would have to check.


For performance and stability reasons, since these are becoming a part of the container's status we may want to consider putting these fields in the container Status struct https://github.com/containerd/containerd/blob/main/pkg/cri/store/container/status.go#L100

Reply:
still thinking these fields should be stored in container status.. as opposed to unmarshaling them out of the spec..

Reply:
Just to make sure I understand correctly, you are suggesting we add all the fields in Resources (CpuPeriod etc) to the container Status struct. Upon container creation [1], we populate all the fields with the values from spec, and during UpdateContainerResources [2], we update both spec and status.

In that case, the resources fields in status will be always the same as those in spec, right? What would be the benefit of storing these fields in status?

[1] https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create.go#L263-L264

[2] https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_update_resources.go#L79

Reply:
right, the benefit would be performance and stability, remove the "extra" db retrieval from the different store and extra unmarshall (specs can be very large)...

Reply:
note changes underway over here: https://github.com/containerd/containerd/pull/7061